source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2478 @@
+KMS) and stored within a document in the MongoDB key vault collection (see
+[Key vault collection schema for data keys](#key-vault-collection-schema-for-data-keys) for a description of the data
+key document). Therefore, a client needs access to both MongoDB and the external KMS service to utilize a data key.
+
+MongoDB key vault collection\
+A MongoDB collection designated to contain data keys. This can either be co-located with
+the data-bearing cluster, or in a separate external MongoDB cluster.
+
+Key Management Service (KMS)\
+An external service providing fixed-size encryption/decryption. Only data keys are
+encrypted and decrypted with KMS.
+
+KMS providers
+
+> A map of KMS providers to credentials. Configured client-side. Example:
+>
+> ```python
+> kms_providers = {
+>    "aws": {
+>       "accessKeyId": AWS_KEYID,
+>       "secretAccessKey": AWS_SECRET,
+>    },
+>    "local": {
+>       "key": LOCAL_KEK
+>    },
+> }
+> ```
+
+KMS provider\
+A configured KMS. Identified by a key in the KMS providers map. The key has the form "<KMS provider type>"
+or "<KMS provider type>:<KMS
+provider name>". Examples: "aws" or "aws:myname". In [libmongocrypt](#libmongocrypt), the
+key is referred to as the KMS ID.
+
+KMS provider type\
+The type of backing KMS. Identified by the string: "aws", "azure", "gcp", "kmip", or "local".
+
+KMS provider name\
+An optional name to identify a KMS provider. Enables configuring multiple KMS providers with the same
+KMS provider type (e.g. "aws:name1" and "aws:name2" can refer to different AWS accounts).
+
+Customer Master Key (CMK)\
+The underlying key AWS KMS uses to encrypt and decrypt. See
+[AWS Key Management Service Concepts](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#master_keys).
+
+schema\
+A MongoDB JSON Schema (either supplied by the server or client-side) which may include metadata about encrypted
+fields. This is a JSON Schema based on draft 4 of the JSON Schema specification,
+[as documented in the MongoDB manual.](https://www.mongodb.com/docs/manual/reference/operator/query/jsonSchema/).
+
+[libmongocrypt](#libmongocrypt)\
+A library, written in C, that coordinates communication, does encryption/decryption,
+caches key and schemas. [Located here](https://github.com/mongodb/libmongocrypt).
+
+[mongocryptd](#mongocryptd)\
+A local process the driver communicates with to determine how to encrypt values in a
+command.
+
+[crypt_shared](#crypt_shared)\
+This term, spelled in all-lowercase with an underscore, refers to the client-side
+field-level-encryption dynamic library provided as part of a MongoDB Enterprise distribution. It replaces
+[mongocryptd](#mongocryptd) as the method of
+`marking-up a database command for encryption <subtype6.intent-to-encrypt>`.
+
+See also:
+
+> - [Introduction on crypt_shared](#crypt_shared)
+> - \[Enabling crypt_shared\](#Enabling crypt_shared)
```

Comment:
The "Enabling crypt_shared" links are broken throughout this document.

Also, the block-quoting probably isn't necessary here.

Reply:
Fixed

Reply:
Re-fixed